Sources/SwiftDocC/Semantics/Metadata/Availability.swift

```diff
@@ -0,0 +1,138 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import Markdown
+
+/// A directive that sets the platform availability information for a documentation page.
+///
+/// `@Available` is analagous to the `@available` attribute in Swift: It allows you to specify a
+/// platform version that the page relates to. To specify a platform and version, list the platform
+/// name and use the `introduced` argument:
+///
+/// ```markdown
+/// @Available(macOS, introduced: "12.0")
+/// ```
+///
+/// The available platforms are `macOS`, `iOS`, `watchOS`, and `tvOS`.
+///
+/// This directive is available on both articles and documentation extension files. In extension
+/// files, the information overrides any information from the symbol itself.
+///
+/// This directive is only valid within a ``Metadata`` directive:
+///
+/// ```markdown
+/// @Metadata {
+///     @Available(macOS, introduced: "12.0")
+/// }
+/// ```
+public final class MetadataAvailability: Semantic, AutomaticDirectiveConvertible {
+    static public let directiveName: String = "Available"
+
+    public enum Platform: String, RawRepresentable, CaseIterable, DirectiveArgumentValueConvertible {
+        case any = "*"
+        case macOS, iOS, watchOS, tvOS
+
+        public init?(rawValue: String) {
+            for platform in Self.allCases {
+                if platform.rawValue.lowercased() == rawValue.lowercased() {
+                    self = platform
+                    return
+                }
+            }
+            return nil
+        }
+    }
+
+    /// The platform that this argument's information applies to.
+    @DirectiveArgumentWrapped(name: .unnamed)
+    public var platform: Platform = .any
+
+    /// The platform version that this page applies to.
+    @DirectiveArgumentWrapped
+    public var introduced: String? = nil
+
+    /// Whether to mark this page as "Deprecated".
+    @DirectiveArgumentWrapped
+    public var isDeprecated: Bool = false
+
+    /// Whether to mark this page as "Beta".
+    @DirectiveArgumentWrapped
+    public var isBeta: Bool = false
```

Comment:
I'm wondering if we should even allow the parsing of these yet since they're unsupported? The chances of 5.8 shipping with this directive but without the DocC-Render support seem non-zero and that would lead to a pretty confusing user experience.

Reply:
Primarily the reason i kept them is that i'd already written the "duplicate argument" warnings before i'd realized that there would need to be more work to support them! 😅 I can take them out (and thus also all the test code surrounding them) which would drop a fair amount of code from this PR.

Reply:
I went ahead and dropped the `isBeta`/`isDeprecated` properties and removed a bunch of code that was interacting with them. This also means i dropped the `*` platform, since it was only there in the first place to support those arguments.